lib/fennec/udp/worker.ex

```diff
@@ -34,15 +35,20 @@ defmodule Fennec.UDP.Worker do
     GenServer.cast(pid, {:process_data, data})
   end
 
-  def start_link(dispatcher, socket, ip, port) do
-    GenServer.start_link(__MODULE__, [dispatcher, socket, ip, port])
+  def start_link(server_opts, dispatcher, socket, ip, port) do
+    GenServer.start_link(__MODULE__, [server_opts, dispatcher, socket, ip, port])
   end
 
   ## GenServer callbacks
 
-  def init([dispatcher, socket, ip, port]) do
+  def init([server_opts, dispatcher, socket, ip, port]) do
     _ = Dispatcher.register_worker(dispatcher, self(), ip, port)
-    {:ok, %{socket: socket, ip: ip, port: port, turn: %TURN{}}}
+    turn_state = %TURN{
+      nonce: "dasda",
+      realm: Keyword.get(server_opts, :realm, "localhost")
```

Comment:
I wanted to explicitly distinguish between protocol state and the worker state. Somehow I would really like to avoid passing entire bare worker state across the entire code.